capellambse/helpers.py

```diff
@@ -116,6 +127,50 @@ def normalize_pure_path(
     return pathlib.PurePosixPath(*parts)
 
 
+if sys.platform.startswith("win"):
+
+    @contextlib.contextmanager
+    def flock(file: pathlib.Path) -> t.Iterator[None]:
+        file = file.resolve()
+        logged = False
+        with self.__repo.joinpath("capellambse.lock").open("wb") as lock:
+            while True:
+                try:
+                    msvcrt.locking(lock.fileno(), msvcrt.LK_LOCK, 1)
+                except OSError as err:
+                    if err.errno == errno.EDEADLOCK:
+                        if not logged:
+                            LOGGER.debug("Waiting for lock file %s", file)
+                            logged = True
+                        time.sleep(1)
+                    else:
+                        raise
+                else:
+                    break
+
+            try:
+                yield
+            finally:
+                msvcrt.locking(lock.fileno(), msvcrt.LK_UNLCK, 1)
+
+else:
+
+    @contextlib.contextmanager
+    def flock(file: pathlib.Path) -> t.Iterator[None]:
+        file = file.resolve()
+        with file.open("wb") as lock:
+            try:
+                fcntl.flock(lock, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            except OSError as err:
```

Comment:
Can't we just use `BlockingIOError` here? `errno.EAGAIN` should always fall under `BlockingIOError` and all other errors are raised anyway.

Reply:
You're right, good catch!